### DISCUSSION TITLE
Using custom history object with createBrowserHistory?

### DISCUSSION THREAD

**Discussion by tryggvigy** [remix-run/react-router#9414] [Q&A]
Hey folks, FAQ has an answer for [How do I access the history object outside of components](https://github.com/remix-run/react-router/blob/59b319feaa12745a434afdef5cadfcabd01206f9/FAQ.md#how-do-i-access-the-history-object-outside-of-components) if you are using  `<BrowserRouter>`, `<HashRouter>`, `<MemoryRouter>`,  `<NativeRouter>`. 

I would like to use one of the routers that support the new v6.4 data APIs: https://reactrouter.com/en/main/routers/picking-a-router#using-v64-data-apis (like `createBrowserRouter`). 

Am I able to use `history` outside of React components with `createBrowserRouter` etc. ? 

From looking at the [source code](https://github.com/remix-run/react-router/blob/release-6.4.2-pre.1/packages/react-router-dom/index.tsx#L197-L211) I can't find any escape hatches to pass the history into these v6.4 route creation functions. `createRouter` is not exported from `@remix-run/router` through `react-router-dom` is it?

**Comment by ryanflorence** [comment]
https://github.com/remix-run/react-router/issues/9422#issuecomment-1302564759
